cmd/kanikoExecute.go

```diff
@@ -74,11 +74,37 @@ func runKanikoExecute(config *kanikoExecuteOptions, telemetryData *telemetry.Cus
 	}
 
 	dockerConfig := []byte(`{"auths":{}}`)
+
+	// respect user provided docker config json file
 	if len(config.DockerConfigJSON) > 0 {
 		var err error
 		dockerConfig, err = fileUtils.FileRead(config.DockerConfigJSON)
 		if err != nil {
-			return errors.Wrapf(err, "failed to read file '%v'", config.DockerConfigJSON)
+			return errors.Wrapf(err, "failed to read existing docker config json at '%v'", config.DockerConfigJSON)
+		}
+	}
+
+	// if : user provided docker config json and registry credentials present then enahance the user provided docker provided json with the registry credentials
+	// else if : no user provided docker config json then create a new docker config json for kaniko
+	if len(config.DockerConfigJSON) > 0 && len(config.ContainerRegistryURL) > 0 && len(config.ContainerRegistryPassword) > 0 && len(config.ContainerRegistryUser) > 0 {
+		targetConfigJson, err := docker.CreateDockerConfigJSON(config.ContainerRegistryURL, config.ContainerRegistryUser, config.ContainerRegistryPassword, "", config.DockerConfigJSON, fileUtils)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update existing docker config json file '%v'", config.DockerConfigJSON)
+		}
+
+		dockerConfig, err = fileUtils.FileRead(targetConfigJson)
+		if err != nil {
+			return errors.Wrapf(err, "failed to read enhanced file '%v'", config.DockerConfigJSON)
+		}
+	} else if len(config.DockerConfigJSON) == 0 && len(config.ContainerRegistryURL) > 0 && len(config.ContainerRegistryPassword) > 0 && len(config.ContainerRegistryUser) > 0 {
+		targetConfigJson, err := docker.CreateDockerConfigJSON(config.ContainerRegistryURL, config.ContainerRegistryUser, config.ContainerRegistryPassword, "", "/kaniko/.docker/config.json", fileUtils)
```

Comment:
minor thing: for this case we essentially don't need line 111 to run if I am not mistaken.
On the other side this should not harm, I guess.

Reply:
yes the file gets written in the package docker, wanted to keep the code a bit consistent when reading , hence added the additional file reading